fix start offset

surfdude75 opened the pull request:
### This PR will...

Fix start offset

### Why is this Pull Request needed?

start offset is not working

### Are there any points in the code the reviewer needs to double check?

### Resolves issues:

### Checklist

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] API or design changes are documented in API.md

robwalch commented:
@surfdude75 under what circumstances? Have you tested both Live and VOD streams?

OrenMe commented:
@surfdude75 im also curious to know when you encountered this and how does this fixes it.
We also encountered start offset not working with vod but haven’t gotten around to debug it yet.

surfdude75 commented:
I have an application that create a HLS feed that works like a like stream, but I need to skip a small interval of seconds in the first chunk. and HLS.js was ignoring this directive.
I verified that the code of code that uses the offset parameter (newDetails.startTimeOffset) is inside an if block that is proceed only if the value of startPosition is -1. But I was getting 0 and it was ignoring that offset parameter. Link bellow to the if block.
 
https://github.com/video-dev/hls.js/blob/9fa8f0aa45c888fe9fe3600659859ffd2bdfe3dc/src/controller/stream-controller.js#L793

So I adjusted the startPosition So it get -1. 

As tests were ok and it worked perfectly for may application I moved on.

Aloha,

Rafael Sobral

robwalch commented on src/controller/stream-controller.js:
```diff
@@ -68,6 +68,7 @@ class StreamController extends BaseStreamController {
         // set new level to playlist loader : this will trigger start level load
         // hls.nextLoadLevel remains until it is set to a new value or until a new frag is successfully loaded
         this.level = hls.nextLoadLevel = startLevel;
+        if (startPosition === -1) startPosition = this.startPosition;
```
This could prevent the block below from ever running:
`if (lastCurrentTime > 0 && startPosition === -1) {`

surfdude75 replied:
preparing an example to check why I needed it and posting tomorrow.

surfdude75 commented on src/controller/stream-controller.js:
```diff
@@ -714,7 +715,7 @@ class StreamController extends BaseStreamController {
     this.fragmentTracker.removeAllFragments();
     this.stalled = false;
-    this.startPosition = this.lastCurrentTime = 0;
+    this.startPosition = this.lastCurrentTime = -1;
```
I do not using start position. I was trusting it would respect the offset in source.
You are right. Thanks.

michaelcunningham19 commented:
I believe due to inactivity and clarity from the author, we can close this PR
Thoughts @robwalch ?

robwalch commented:
> I believe due to inactivity and clarity from the author, we can close this PR
@michaelcunningham19 Yes.  Sounds like a configuration issue. If the need comes up we can always reopen.
